psi4/driver/procrouting/proc.py

```diff
@@ -1228,10 +1228,31 @@ def scf_wavefunction_factory(name, ref_wfn, reference, **kwargs):
             wfn.set_sad_fitting_basissets(sad_fitting_list)
             optstash.restore()
 
-    # Deal with the EXTERN issues
+    if hasattr(core, "EXTERN") and 'external_potentials' in kwargs: 
+        core.print_out("\n  Warning! Both an external potential EXTERN object and the external_potential" +
+                       " keyword argument are specified. The external_potential keyword argument will be ignored.\n")
+
+    # If EXTERN is set, then place that potential on the wfn
     if hasattr(core, "EXTERN"):
+        wfn.set_external_potential_c(core.EXTERN) # This is for the FSAPT procedure
         wfn.set_external_potential(core.EXTERN)
 
+    elif 'external_potentials' in kwargs:
+        # For FSAPT, we can take a dictionary of external potentials, e.g.,
+        # external_potentials={'A': potA, 'B': potB, 'C': potC} (any optional)
+        # For the dimer SAPT calculation, we need to account for the external potential
+        # in all of the subsystems A, B, C. So we add them all in total_external_potential
+        # and set the external potential to the dimer wave function
+        total_external_potential = core.ExternalPotential()
+
+        ext_pot_type = [wfn.set_external_potential_a, wfn.set_external_potential_b, wfn.set_external_potential_c]
```

Comment:
I recommend finding the setter function corresponding to a given letter with `getattr` instead. `getattr(wfn, f"set_external_potential_{frag.lower()}")`

Reply:
Fixed.